src/components/submission-common.test.js

```diff
@@ -269,6 +281,71 @@ describe('requiredInputsWithoutSource', () => {
       validBooleanInput
     ]
 
-    expect(requiredInputsWithoutSource(inputs).length).toBe(0)
+    expect(validateInputs(inputs, emptyDataTableAttributes).length).toBe(0)
+  })
+
+  const arrayInput = (name, arrayType, value) => {
+    return {
+      input_name: name,
+      input_type: {
+        type: 'optional',
+        optional_type: {
+          type: 'array',
+          array_type: {
+            type: 'primitive',
+            primitive_type: arrayType
+          }
+        }
+      },
+      source: {
+        type: 'literal',
+        parameter_value: value
+      }
+    }
+  }
+
+  it('should validate array literals', () => {
+    const inputsWithArraysDefinition = [
+      // int arrays (float similar enough that not worried about specifics)
+      arrayInput('validInt', 'Int', [1, 2, 3]), // success
+      arrayInput('validIntString', 'Int', ['1', 2, 3]), // success
+      arrayInput('validIntStringArray', 'Int', '[1, 2, 3]'), // success
+      arrayInput('validIntStringSingleton', 'Int', '1'), // info
+      arrayInput('validIntSingleton', 'Int', 1), // info
+      arrayInput('invalidIntEmpty', 'Int', ''), // error
+      arrayInput('invalidIntFloat', 'Int', [1, 2.5, 3]), // error
+      arrayInput('invalidIntString', 'Int', [1, 'a', 3]), // error
+
+      // boolean arrays
+      arrayInput('validBoolean', 'Boolean', [true, false, true]), // success
+      arrayInput('validBooleanString', 'Boolean', ['true', false, true]), // success
+      arrayInput('validBooleanStringArray', 'Boolean', '[true, false, true]'), // success
+      arrayInput('validBooleanStringSingleton', 'Boolean', 'true'), // info
+      arrayInput('validBooleanSingleton', 'Boolean', true), // info
+      arrayInput('invalidBooleanEmpty', 'Boolean', ''), // error
+      arrayInput('invalidBooleanInt', 'Boolean', [true, 2, false]), // error
+      arrayInput('invalidBooleanString', 'Boolean', [true, 'a', false]), // error
+
+      // string arrays
+      arrayInput('validString', 'String', ['a', 'b', 'c']), // success
+      arrayInput('validStringArray', 'String', '["a", "b", "c"]'), // success
+      arrayInput('validStringSingleton', 'String', 'foo'), // info
+      arrayInput('invalidStringEmpty', 'String', '') // error
+    ]
+
+    const inputMessages = validateInputs(inputsWithArraysDefinition, emptyDataTableAttributes)
```

Comment:
TOL: I wonder whether this is really `validations` now instead of `messages`? You could argue the main output is the set of enum values we use to decide "is this input invalid" (to disable the submit button), and that the messages are incidental. It might also feel better to have some kind of active "All good, no further comment" value if this is called "validations" instead of "messages"